logging_middleware.go

```diff
@@ -3,21 +3,54 @@ package handlers
 import (
 	"context"
 	"net/http"
+	"regexp"
 	"time"
 
+	"gopkg.in/errgo.v1"
+
 	"github.com/codegangsta/negroni"
 	"github.com/sirupsen/logrus"
 )
 
+var (
+	loggerFuncMap = map[logrus.Level]func(logrus.FieldLogger, string, ...interface{}){
+		logrus.DebugLevel: logrus.FieldLogger.Debugf,
+		logrus.InfoLevel:  logrus.FieldLogger.Infof,
+		logrus.WarnLevel:  logrus.FieldLogger.Warnf,
+		logrus.ErrorLevel: logrus.FieldLogger.Errorf,
+		logrus.FatalLevel: logrus.FieldLogger.Fatalf,
+		logrus.PanicLevel: logrus.FieldLogger.Panicf,
+	}
+)
+
+type patternInfo struct {
+	re    *regexp.Regexp
+	level logrus.Level
+}
+
 type LoggingMiddleware struct {
-	logger logrus.FieldLogger
+	logger  logrus.FieldLogger
+	filters map[string]patternInfo
```

Comment:
Why making this a map if you are always using this like an array ?

Reply:
True, it doesn't make a lot of sense

Reply:
done